doc/man/man3/UI.3
.TH "UI" 3 "Thu Jun 9 2011" "Version 0.9" "YATD" \" -*- nroff -*-
.ad l
.nh
.SH NAME
UI \- 
.PP
Classe de l'interface graphique.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <UI.h>\fP
.SS "Connecteurs publics"

.in +1c
.ti -1c
.RI "void \fBsetWaveName\fP (QString \fBname\fP)"
.br
.RI "\fIChangement de vague. \fP"
.ti -1c
.RI "void \fBaddCred\fP ()"
.br
.RI "\fIGain d'un crédit. \fP"
.ti -1c
.RI "void \fBloseLife\fP ()"
.br
.RI "\fIPerte de vie. \fP"
.ti -1c
.RI "void \fBstartWave\fP ()"
.br
.RI "\fIVague suivante. \fP"
.ti -1c
.RI "void \fBbuyWaterTower\fP ()"
.br
.RI "\fIAchat d'une défense 'pistolet à eau'. \fP"
.ti -1c
.RI "void \fBbuySlingshotTower\fP ()"
.br
.RI "\fIAchat d'une défense 'lance-pierres'. \fP"
.ti -1c
.RI "void \fBbuyPaintballTower\fP ()"
.br
.RI "\fIAchat d'une défense 'paintball'. \fP"
.ti -1c
.RI "void \fBbuyBowlingTower\fP ()"
.br
.RI "\fIAchat d'une défense 'pétanque'. \fP"
.ti -1c
.RI "void \fBselectTower\fP (\fBTower\fP *\fBtower\fP)"
.br
.RI "\fISélection d'une tour. \fP"
.ti -1c
.RI "void \fBsellSelectedTower\fP ()"
.br
.RI "\fIVente de la tour sélectionnée. \fP"
.ti -1c
.RI "void \fBupgradeSelectedTower\fP ()"
.br
.RI "\fIAmélioration de la défense sélectionnée. \fP"
.ti -1c
.RI "void \fBtogglePause\fP ()"
.br
.RI "\fIPause. \fP"
.in -1c
.SS "Signaux"

.in +1c
.ti -1c
.RI "void \fBbuyTower\fP (QString type)"
.br
.RI "\fIAchat d'une tour. \fP"
.ti -1c
.RI "void \fBnextWave\fP ()"
.br
.RI "\fIVague suivante. \fP"
.ti -1c
.RI "void \fBtowerSold\fP (\fBTower\fP *\fBtower\fP)"
.br
.RI "\fIVente d'une défense. \fP"
.ti -1c
.RI "void \fBdefeat\fP ()"
.br
.RI "\fISignal de défaite. \fP"
.ti -1c
.RI "void \fBpause\fP ()"
.br
.RI "\fIPause. \fP"
.ti -1c
.RI "void \fBunpause\fP ()"
.br
.RI "\fIUnpause. \fP"
.in -1c
.SS "Fonctions membres publiques"

.in +1c
.ti -1c
.RI "\fBUI\fP (QWidget *parent)"
.br
.RI "\fIConstructeur de \fBUI\fP. \fP"
.in -1c
.SS "Champs de données"

.in +1c
.ti -1c
.RI "QPushButton * \fBstart\fP"
.br
.RI "\fILe bouton pour lancer la vague suivante. \fP"
.in -1c
.SS "Attributs privés"

.in +1c
.ti -1c
.RI "QGroupBox * \fBtower\fP"
.br
.RI "\fIRegroupe les éléments du magasin de défenses. \fP"
.ti -1c
.RI "QGroupBox * \fBstats\fP"
.br
.RI "\fIRegroupe les éléments ayant attrait à la défense sélectionnée. \fP"
.ti -1c
.RI "QPushButton * \fBpauseBtn\fP"
.br
.RI "\fILe bouton pour mettre le jeu en pause. \fP"
.ti -1c
.RI "QPushButton * \fBsell\fP"
.br
.RI "\fILe bouton pour vendre la tour sélectionnée. \fP"
.ti -1c
.RI "QPushButton * \fBupg\fP"
.br
.RI "\fILe bouton pour améliorer la tour sélectionnée. \fP"
.ti -1c
.RI "QPushButton * \fBwater\fP"
.br
.RI "\fILe bouton pour acheter un pistolet à eau. \fP"
.ti -1c
.RI "QPushButton * \fBsling\fP"
.br
.RI "\fILe bouton pour acheter un lance-pierre. \fP"
.ti -1c
.RI "QPushButton * \fBbowling\fP"
.br
.RI "\fILe bouton pour engager un joueur de pétanque. \fP"
.ti -1c
.RI "QPushButton * \fBpaintball\fP"
.br
.RI "\fILe bouton pour acheter un \fBPaintball\fP. \fP"
.ti -1c
.RI "QLabel * \fBname\fP"
.br
.RI "\fILe texte 'Type : '. \fP"
.ti -1c
.RI "QLabel * \fBlvl\fP"
.br
.RI "\fILe texte 'Niveau : '. \fP"
.ti -1c
.RI "QLabel * \fBrange\fP"
.br
.RI "\fILe texte 'portée : '. \fP"
.ti -1c
.RI "QLabel * \fBfirerate\fP"
.br
.RI "\fILe texte 'cadence : '. \fP"
.ti -1c
.RI "QLabel * \fBt_name\fP"
.br
.RI "\fILe type de la tour sélectionnée. \fP"
.ti -1c
.RI "QLabel * \fBt_lvl\fP"
.br
.RI "\fILe niveau de la tour sélectionnée. \fP"
.ti -1c
.RI "QLabel * \fBt_range\fP"
.br
.RI "\fILa portée de la tour sélectionnée. \fP"
.ti -1c
.RI "QLabel * \fBt_firerate\fP"
.br
.RI "\fILa cadence de tir de la tour sélectionnée. \fP"
.ti -1c
.RI "QLabel * \fBwave\fP"
.br
.RI "\fIle nom de la vague en cours. \fP"
.ti -1c
.RI "QLabel * \fBcred_txt\fP"
.br
.RI "\fILe texte 'crédits : '. \fP"
.ti -1c
.RI "QLabel * \fBlife_txt\fP"
.br
.RI "\fILe texte 'vie : '. \fP"
.ti -1c
.RI "QLCDNumber * \fBcred\fP"
.br
.RI "\fIL'affciheur de crédits restants. \fP"
.ti -1c
.RI "QLCDNumber * \fBlife\fP"
.br
.RI "\fIL'afficheur de vie restantes. \fP"
.ti -1c
.RI "\fBTower\fP * \fBselected\fP"
.br
.RI "\fIUn pointeur vers la tour sélectionnée actuellement. \fP"
.ti -1c
.RI "bool \fBpauseVar\fP"
.br
.RI "\fITrue si le jeu est en pause, False sinon. \fP"
.in -1c
.SH "Description détaillée"
.PP 
Gère l'ensemble des éléments de l'interface graphique affichée à droite. 
.SH "Documentation des constructeurs et destructeur"
.PP 
.SS "UI::UI (QWidget *parent)"initialise l'interface utilisateur. 
.PP
\fBParamètres:\fP
.RS 4
\fIparent\fP Le widge parent de l'UI. 
.RE
.PP

.SH "Documentation des fonctions membres"
.PP 
.SS "void UI::addCred ()\fC [slot]\fP"Incrémente de 1 les crédit du joueur. 
.SS "void UI::buyBowlingTower ()\fC [slot]\fP"Vérifie si le joueur possède suffisament de crédit puis décompte le prix. 
.SS "void UI::buyPaintballTower ()\fC [slot]\fP"Vérifie si le joueur possède suffisament de crédit puis décompte le prix. 
.SS "void UI::buySlingshotTower ()\fC [slot]\fP"Vérifie si le joueur possède suffisament de crédit puis décompte le prix. 
.SS "void UI::buyTower (QStringtype)\fC [signal]\fP"Indique au \fBRender\fP de placer la défense achetée. 
.PP
\fBParamètres:\fP
.RS 4
\fItype\fP Le type de la défense achetée. 
.RE
.PP

.SS "void UI::buyWaterTower ()\fC [slot]\fP"Vérifie si le joueur possède suffisament de crédit puis décompte le prix. 
.SS "void UI::defeat ()\fC [signal]\fP"Indique au programme de stopper le jeu. 
.SS "void UI::loseLife ()\fC [slot]\fP"Décrémente de 1 le total de vies du joueur. 
.SS "void UI::nextWave ()\fC [signal]\fP"Indique au \fBRender\fP de lancer la vague suivante. 
.SS "void UI::pause ()\fC [signal]\fP"Arrête le jeu en cours. 
.SS "void UI::selectTower (\fBTower\fP *tower)\fC [slot]\fP"Affiche les caractérisque de la défense sélectionnée. 
.PP
\fBParamètres:\fP
.RS 4
\fItower\fP Un pointeur vers la nouvelle défense sélectionnée. 
.RE
.PP

.SS "void UI::sellSelectedTower ()\fC [slot]\fP"Reçoit le signal du bouton 'Vendre' et recrédite le joueur du moitié du prix de la défense. 
.SS "void UI::setWaveName (QStringname)\fC [slot]\fP"Change le nom de la vague courante. 
.PP
\fBParamètres:\fP
.RS 4
\fIname\fP Le nom de la nouvelle vague. 
.RE
.PP

.SS "void UI::startWave ()\fC [slot]\fP"Reçoit le signal du bouton 'lancer la vague suivante'. 
.SS "void UI::togglePause ()\fC [slot]\fP"Arrête le jeu en cours ou relance le jeu en pause. 
.SS "void UI::towerSold (\fBTower\fP *tower)\fC [signal]\fP"Indique au \fBRender\fP de détruire une défense vendue. 
.PP
\fBParamètres:\fP
.RS 4
\fItower\fP Un pointeur vers la défense à détruire. 
.RE
.PP

.SS "void UI::unpause ()\fC [signal]\fP"Relance le jeu. 
.SS "void UI::upgradeSelectedTower ()\fC [slot]\fP"Reçoit le signal du bouton 'améliorer' et améliore la tour si les crédits sont suffisants. 
.SH "Documentation des champs"
.PP 
.SS "QPushButton* \fBUI::bowling\fP\fC [private]\fP"
.SS "QLCDNumber* \fBUI::cred\fP\fC [private]\fP"
.SS "QLabel* \fBUI::cred_txt\fP\fC [private]\fP"
.SS "QLabel* \fBUI::firerate\fP\fC [private]\fP"
.SS "QLCDNumber* \fBUI::life\fP\fC [private]\fP"
.SS "QLabel* \fBUI::life_txt\fP\fC [private]\fP"
.SS "QLabel* \fBUI::lvl\fP\fC [private]\fP"
.SS "QLabel* \fBUI::name\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::paintball\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::pauseBtn\fP\fC [private]\fP"
.SS "bool \fBUI::pauseVar\fP\fC [private]\fP"
.SS "QLabel* \fBUI::range\fP\fC [private]\fP"
.SS "\fBTower\fP* \fBUI::selected\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::sell\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::sling\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::start\fP"
.SS "QGroupBox* \fBUI::stats\fP\fC [private]\fP"
.SS "QLabel* \fBUI::t_firerate\fP\fC [private]\fP"
.SS "QLabel* \fBUI::t_lvl\fP\fC [private]\fP"
.SS "QLabel* \fBUI::t_name\fP\fC [private]\fP"
.SS "QLabel* \fBUI::t_range\fP\fC [private]\fP"
.SS "QGroupBox* \fBUI::tower\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::upg\fP\fC [private]\fP"
.SS "QPushButton* \fBUI::water\fP\fC [private]\fP"
.SS "QLabel* \fBUI::wave\fP\fC [private]\fP"

.SH "Auteur"
.PP 
Généré automatiquement par Doxygen pour YATD à partir du code source.


src/UI.cpp
#include <QString>

#include "UI.h"
#include "define.h"

UI::UI(QWidget * parent) : QWidget(parent), selected(NULL), pauseVar(false)
{
    tower = new QGroupBox("Créer une défense", this);
    tower->setGeometry(0, 120, UI_WIDTH, 180);
    stats = new QGroupBox("Stats", this);
    stats->setGeometry(0, 310, UI_WIDTH, 170);

    pauseBtn = new QPushButton("Pause", this);
    pauseBtn->setGeometry(10, 485, 180, 25);
    QObject::connect(pauseBtn, SIGNAL(clicked()), this, SLOT(togglePause()));
    start = new QPushButton("Lancer la vague suivante", this);
    start->setGeometry(10, 30, 180, 25);
    QObject::connect(start, SIGNAL(clicked()), this, SLOT(startWave()));
    upg = new QPushButton("Améliorer", stats);
    upg->setGeometry(20, 140, 75, 20);
    QObject::connect(upg, SIGNAL(clicked()), this, SLOT(upgradeSelectedTower()));
    sell = new QPushButton("Vendre", stats);
    sell->setGeometry(105, 140, 75, 20);
    QObject::connect(sell, SIGNAL(clicked()), this, SLOT(sellSelectedTower()));

    // Tower shop
    water = new QPushButton("Eau", tower);
    water->setGeometry(10, 25, (UI_WIDTH - 30)/2, 65);
    QObject::connect(water, SIGNAL(clicked()), this, SLOT(buyWaterTower()));
    sling = new QPushButton("Pierres", tower);
    sling->setGeometry((UI_WIDTH - 30)/2 + 20, 25, (UI_WIDTH - 30)/2, 65);
    QObject::connect(sling, SIGNAL(clicked()), this, SLOT(buySlingshotTower()));
    bowling = new QPushButton("Pétanque", tower);
    bowling->setGeometry(10, 100, (UI_WIDTH - 30)/2, 65);
    QObject::connect(bowling, SIGNAL(clicked()), this, SLOT(buyBowlingTower()));
    paintball = new QPushButton("Paintball", tower);
    paintball->setGeometry((UI_WIDTH - 30)/2 + 20, 100, (UI_WIDTH - 30)/2, 65);
    QObject::connect(paintball, SIGNAL(clicked()), this, SLOT(buyPaintballTower()));

    //Tower Stat
    name = new QLabel("type de la tour : ", stats);
    name->setGeometry(5, 15, 100, 20);
    lvl = new QLabel("niveau : ", stats);
    lvl->setGeometry(5, 40, 100, 20);
    range = new QLabel("portée : ", stats);
    range->setGeometry(5, 65, 100, 20);
    firerate = new QLabel("cadence de tir : ", stats);
    firerate->setGeometry(5, 90, 100, 20);

    t_name = new QLabel("", stats);
    t_name->setGeometry(130, 15, 100, 20);
    t_lvl = new QLabel("", stats);
    t_lvl->setGeometry(130, 40, 100, 20);
    t_range = new QLabel("", stats);
    t_range->setGeometry(130, 65, 100, 20);
    t_firerate = new QLabel("", stats);
    t_firerate->setGeometry(130, 90, 100, 20);

    // Game Stats display
    wave = new QLabel("", this);
    wave->setGeometry(0, 5, UI_WIDTH, 20);
    cred_txt = new QLabel("Crédits :", this);
    cred_txt->move(0, 65);
    life_txt = new QLabel("Vies :", this);
    life_txt->move(0, 95);

    cred = new QLCDNumber(this);
    cred->setSegmentStyle(QLCDNumber::Flat);
    cred->move(135, 65);
    cred->display(START_CRED);
    life = new QLCDNumber(this);
    life->setSegmentStyle(QLCDNumber::Flat);
    life->move(135, 95);
    life->display(START_LIFE);
}

void UI::startWave()
{
    emit nextWave();
}

void UI::setWaveName(QString name)
{
    wave->setText(name);
}

void UI::addCred()
{
    cred->display(cred->intValue() + 1);
}

void UI::loseLife()
{
    life->display(life->intValue() - 1);
    if (life->intValue() <= 0)
        emit defeat();
}

void UI::buyWaterTower()
{
    if (cred->intValue() >= WATER_PRICE) {
        cred->display(cred->intValue() - WATER_PRICE);
        emit buyTower(QString("water"));
    }
}

void UI::buySlingshotTower()
{
    if (cred->intValue() >= SLINGSHOT_PRICE) {
        cred->display(cred->intValue() - SLINGSHOT_PRICE);
        emit buyTower(QString("slingshot"));
    }
}

void UI::buyPaintballTower()
{
    if (cred->intValue() >= PAINTBALL_PRICE) {
        cred->display(cred->intValue() - PAINTBALL_PRICE);
        emit buyTower(QString("paintball"));
    }
}

void UI::buyBowlingTower()
{
    if (cred->intValue() >= BOWLING_PRICE) {
        cred->display(cred->intValue() - BOWLING_PRICE);
        emit buyTower(QString("bowling"));
    }
}

void UI::selectTower(Tower * tower)
{
    selected = tower;
    t_name->setText(tower->getType());
    t_lvl->setText(QString().setNum(tower->getLvl()));
    t_range->setText(QString().setNum(tower->getRange()));
    t_firerate->setText(QString().setNum(tower->getFirerate()));
}

void UI::sellSelectedTower()
{
    if (selected != NULL) {
        cred->display(cred->intValue() + (int) selected->getPrice());
        emit towerSold(selected);
        t_name->setText("");
        t_lvl->setText("");
        t_range->setText("");
        t_firerate->setText("");
        selected == NULL;
    }
}

void UI::upgradeSelectedTower()
{
    if (selected != NULL)
        if (selected->getLvl() < 3) {
            if (cred->intValue() >= selected->getUpgCost()) {
                cred->display(cred->intValue() - (int) selected->getUpgCost());
                selected->upgrade();
                t_name->setText(selected->getType());
                t_lvl->setText(QString().setNum(selected->getLvl()));
                t_range->setText(QString().setNum(selected->getRange()));
                t_firerate->setText(QString().setNum(selected->getFirerate()));
            }
        }
}

void UI::togglePause()
{
    if (pauseVar) {
        emit unpause();
        pauseVar = false;
    } else {
        emit pause();
        pauseVar = true;
    }
}
